internal/services/databricks/databricks_virtual_network_peering_resource.go

```diff
@@ -0,0 +1,356 @@
+	allowVirtualNetworkAccess := d.Get("allow_virtual_network_access").(bool)
+	useRemoteGateways := d.Get("use_remote_gateways").(bool)
+	remoteVirtualNetwork := d.Get("remote_virtual_network_id").(string)
+	databricksAddressSpace := utils.ExpandStringSlice(d.Get("address_space_prefixes").([]interface{}))
+	remoteAddressSpace := utils.ExpandStringSlice(d.Get("remote_address_space_prefixes").([]interface{}))
+
+	props := vnetpeering.VirtualNetworkPeeringPropertiesFormat{
+		DatabricksAddressSpace: &vnetpeering.AddressSpace{
+			AddressPrefixes: databricksAddressSpace,
+		},
+		// The RP always creates the same vNet ID for the Databricks internal vNet in the below format:
+		// '/subscriptions/{subscription}/resourceGroups/{group1}/providers/Microsoft.Network/virtualNetworks/workers-vnet'
+		DatabricksVirtualNetwork: &vnetpeering.VirtualNetworkPeeringPropertiesFormatDatabricksVirtualNetwork{
+			Id: utils.String(networkParse.NewVirtualNetworkID(id.SubscriptionId, id.ResourceGroupName, "workers-vnet").ID()),
+		},
+		RemoteAddressSpace: &vnetpeering.AddressSpace{
+			AddressPrefixes: remoteAddressSpace,
+		},
+		RemoteVirtualNetwork: vnetpeering.VirtualNetworkPeeringPropertiesFormatRemoteVirtualNetwork{
+			Id: utils.String(remoteVirtualNetwork),
+		},
+		AllowForwardedTraffic:     &allowForwardedTraffic,
+		AllowGatewayTransit:       &allowGatewayTransit,
+		AllowVirtualNetworkAccess: &allowVirtualNetworkAccess,
+		UseRemoteGateways:         &useRemoteGateways,
+	}
+
+	peer := vnetpeering.VirtualNetworkPeering{
+		Name:       &id.VirtualNetworkPeeringName,
+		Properties: props,
+	}
+
+	locks.ByID(databricksVnetPeeringsResourceType)
+	defer locks.UnlockByID(databricksVnetPeeringsResourceType)
+
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		return fmt.Errorf("internal-error: context had no deadline")
+	}
+
+	stateConf := &pluginsdk.StateChangeConf{
+		Pending: []string{"Pending"},
+		Target:  []string{"Created"},
+		Timeout: time.Until(deadline),
+		Delay:   15 * time.Second,
+		Refresh: func() (interface{}, string, error) {
+			future, err := client.CreateOrUpdate(ctx, id, peer)
+			if err != nil {
+				if utils.ResponseErrorIsRetryable(err) {
+					return future.HttpResponse, "Pending", err
+				} else {
+					if resp := future.HttpResponse; resp != nil && response.WasBadRequest(resp) && strings.Contains(err.Error(), "ReferencedResourceNotProvisioned") {
+						// Resource is not yet ready, this may be the case if the Vnet was just created or another peering was just initiated.
+						return future.HttpResponse, "Pending", err
+					}
+				}
+
+				return future.HttpResponse, "", err
+			}
+
+			if err = future.Poller.PollUntilDone(ctx); err != nil {
+				return future.HttpResponse, "", err
+			}
+
+			return future.HttpResponse, "Created", nil
+		},
+	}
+
+	if _, err := stateConf.WaitForStateContext(ctx); err != nil {
+		return fmt.Errorf("waiting for Databricks %s to be created: %+v", id, err)
+	}
```

Comment:
since we're locking on the creation and we wait for the dependent resources to be created (since we won't return either the Virtual Network or the DataBricks Cluster until it's marked as fully provisioned) - I don't believe this is still needed - so I'm wondering if we can switch to using the regular LRO directly and remove the `WaitForState` func entirely?

```suggestion
	if err := client.CreateOrUpdateThenPoll(ctx, id, peer); err != nil {
		return fmt.Errorf("creating Databricks %s: %+v", id, err)
	}
```